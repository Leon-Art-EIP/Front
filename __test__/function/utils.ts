import type { ReactTestRendererJSON } from "react-test-renderer";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import { expect } from "vitest";

function toJson(component: renderer.ReactTestRenderer): ReactTestRendererJSON {
  const result = component.toJSON();
  expect(result).toBeDefined();

  return result as ReactTestRendererJSON;
}

export function testSnapshot(element: JSX.Element): void {
  const component = renderer.create(element);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
}
