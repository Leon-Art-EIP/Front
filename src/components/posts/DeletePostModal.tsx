import { Button, Modal } from "../lib";

interface IDeletePostModalProps {
  handleClose: () => void;
  handleDelete: () => void;
}

export default function DeletePostModal(props: IDeletePostModalProps): JSX.Element {
  const onCancel = () => {
    props.handleClose();
  };

  const onDelete = () => {
    props.handleDelete();
  };

  return (
    <Modal isOpen handleClose={props.handleClose}>
      <div className="flex flex-col gap-4 p-4 rounded-2xl bg-background max-w-3xl">
        <h2>Voulez-vous vraiment supprimer ce post ?</h2>
        <div className="flex gap-8 justify-center">
          <Button type="button" color="secondary" onClick={onCancel}>
            Annuler
          </Button>
          <Button type="button" color="primary" onClick={onDelete}>
            Oui
          </Button>
        </div>
      </div>
    </Modal>
  );
}
