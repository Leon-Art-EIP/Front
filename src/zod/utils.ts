export const getValueOrUndefined = (arg: unknown) =>
  arg === "" || arg === undefined || arg === null ? undefined : arg;
