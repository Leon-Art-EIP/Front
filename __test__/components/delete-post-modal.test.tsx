import { test } from "vitest";
import { testSnapshot } from "../function/utils";
import DeletePostModal from "../../src/components/posts/DeletePostModal";

test("StatusBadge snapshot - open state", () => {
  testSnapshot(<DeletePostModal handleClose={() => {}} handleDelete={() => {}} />);
});
