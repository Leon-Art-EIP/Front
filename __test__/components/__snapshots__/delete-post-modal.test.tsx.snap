// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StatusBadge snapshot - open state 1`] = `
<div
  aria-labelledby="modal-title"
  aria-modal="true"
  className="fixed z-40 inset-0 overflow-y-auto"
  role="dialog"
>
  <div
    className="flex items-end justify-center min-h-screen pt-4 px-4 pb-20 text-center sm:block sm:p-0"
  >
    <div
      aria-hidden="true"
      className="fixed inset-0 bg-white bg-opacity-80 transition-opacity"
      onClick={[Function]}
    />
    <span
      aria-hidden="true"
      className="hidden sm:inline-block sm:align-middle sm:h-screen"
    >
      â€‹
    </span>
    <div
      className="inline-block align-bottom bg-background-hl text-tertiary rounded-lg text-left overflow-hidden shadow-xl transform transition-all max-w-[95%] sm:my-8 sm:align-middle"
    >
      <div
        className="bg-background text-tertiary px-4 pt-5 pb-4 sm:p-6 sm:pb-4"
      >
        <div
          className="flex flex-col gap-4 p-4 rounded-2xl bg-background max-w-3xl"
        >
          <h2>
            Voulez-vous vraiment supprimer ce post ?
          </h2>
          <div
            className="flex gap-8 justify-center"
          >
            <button
              className="rounded-3xl px-16 py-3 font-semibold text-base transition duration-200 flex items-center justify-center text-tertiary bg-secondary hover:bg-secondary-hover disabled:bg-secondary-disabled disabled: text-secondary undefined"
              onClick={[Function]}
              type="button"
            >
              Annuler
            </button>
            <button
              className="rounded-3xl px-16 py-3 font-semibold text-base transition duration-200 flex items-center justify-center text-white bg-primary hover:bg-primary-hover disabled:bg-primary-disabled undefined"
              onClick={[Function]}
              type="button"
            >
              Oui
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
